sdktests/server_side_persistence_base.go

```diff
@@ -119,58 +134,560 @@ func newServerSidePersistentTests(
+			persistence := NewPersistence()
+			persistence.SetStore(servicedef.SDKConfigPersistentStore{
+				Type: s.persistentStore.Type(),
+				DSN:  s.persistentStore.DSN(),
+			})
+			persistence.SetCache(servicedef.SDKConfigPersistentCache{
+				Mode: servicedef.CacheModeTTL,
+				TTL:  o.Some(1),
+			})
+			context := ldcontext.New("user-key")
+
+			s.runWithEmptyStore(t, "cache hit persists for TTL", func(t *ldtest.T) {
+				client := NewSDKClient(t, persistence)
+
+				require.NoError(t, s.persistentStore.WriteMap(s.defaultPrefix, "features", s.initialFlags))
+
+				pollUntilFlagValueUpdated(t, client, "flag-key", context,
+					ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+
+				// Completely reset the database so there are no valid flag definitions
+				require.NoError(t, s.persistentStore.Reset())
+
+				h.RequireNever(t,
+					checkForUpdatedValue(t, client, "flag-key", context,
+						ldvalue.String("fallthrough"), ldvalue.String("default"), ldvalue.String("default")),
+					time.Millisecond*500, time.Millisecond*20, "flag value was updated before cache TTL")
+
+				h.RequireEventually(t,
+					checkForUpdatedValue(t, client, "flag-key", context,
+						ldvalue.String("fallthrough"), ldvalue.String("default"), ldvalue.String("default")),
+					time.Second, time.Millisecond*20, "flag value was NOT updated after cache TTL")
+			})
+
+			s.runWithEmptyStore(t, "cache miss persists for TTL", func(t *ldtest.T) {
+				client := NewSDKClient(t, persistence)
+
+				result := client.EvaluateFlag(t, servicedef.EvaluateFlagParams{
+					FlagKey:      "flag-key",
+					Context:      o.Some(context),
+					ValueType:    servicedef.ValueTypeAny,
+					DefaultValue: ldvalue.String("default"),
+					Detail:       true,
+				})
+
+				m.In(t).Assert(result.Value, m.Equal(ldvalue.String("default")))
+				m.In(t).Assert(result.Reason.Value().GetErrorKind(), m.Equal(ldreason.EvalErrorFlagNotFound))
+
+				require.NoError(t, s.persistentStore.WriteMap(s.defaultPrefix, "features", s.initialFlags))
+
+				h.RequireNever(t,
+					checkForUpdatedValue(t, client, "flag-key", context,
+						ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default")),
+					time.Microsecond*500, time.Millisecond*20, "flag value was updated before cache TTL")
+
+				h.RequireEventually(t,
+					checkForUpdatedValue(t, client, "flag-key", context,
+						ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default")),
+					time.Second, time.Millisecond*20, "flag value was NOT updated after cache TTL")
+			})
+		})
+
+		s.runWithEmptyStore(t, "caches flag forever", func(t *ldtest.T) {
+			persistence := NewPersistence()
+			persistence.SetStore(servicedef.SDKConfigPersistentStore{
+				Type: s.persistentStore.Type(),
+				DSN:  s.persistentStore.DSN(),
+			})
+			persistence.SetCache(servicedef.SDKConfigPersistentCache{
+				Mode: servicedef.CacheModeInfinite,
```

Comment:
Infinite mode isn't a uniform capability. I guess we can suppress the test though.

Reply:
That's a good point. I could make it a capability, but it does kind of feel like not supporting infinite is more of a suppression thing. If you can support the other two, seems like you should be able to support this.